scripts/ohno.js

```diff
@@ -18,23 +18,16 @@ const ohno = (p) => {
     p.sendMessage('[scarlet]⚠[yellow]Sorry, the max number of ohno units has been reached.');
     return;
   }
-  const existing = []; // Keep track of spawned units to avoid changing wrong ones
-
-  Groups.unit.each((u) => {
-    // loop through all units and save their id
-    if (u.type === UnitTypes.atrax) {
-      existing.push(u.id);
-    }
-  });
-
-  UnitTypes.atrax.spawn(Team.sharded, p.unit().x, p.unit().y); // spawn unit at player's position
+  if(Vars.indexer.findEnemyTile(p.team(), p.unit().x, p.unit().y, 8, () -> true) != null){
+    p.sendMessage('[scarlet]⚠[yellow]Too close to an enemy tile!');
+    return;
+  }
 
-  const targetUnit = Groups.unit.find(
-    // get the unit we just spawned
-    (u) => u.type === UnitTypes.atrax && !existing.includes(u.id)
-  );
+  const targetUnit = UnitTypes.atrax.spawn(Team.sharded, p.unit().x, p.unit().y); // spawn unit at player's position
```

Comment:
I understand how this _should_  replace the above logic, but I think I tested this method when I wrote this and `unit.spawn()` does not return the unit's class. I can play around and test it again but that is likely why the logic above exists.

Reply:
![image](https://user-images.githubusercontent.com/71201189/205603179-21459de7-d23d-480d-a542-37de893cce6e.png)
works in console